rackcity/utils/change_planner_utils.py

```diff
@@ -178,3 +178,117 @@ def get_page_count_response_for_cp(request, change_plan):
+        })
+    if asset_cp.asset_conflict_location:
+        conflicts.append({
+            "conflict_message":
+                conflicting_asset_message_1 + "rack location" +
+                conflicting_asset_message_2 + nonresolvable_message,
+            "conflicting_asset": asset_cp.asset_conflict_location.id,
+            "conflict_resolvable": False,
+        })
+    deleted_relation_message = " associated with this asset has been deleted. "
+    if asset_cp.model is None:
+        conflicts.append({
+            "conflict_message":
+                "The model" + deleted_relation_message + nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.rack is None:
+        conflicts.append({
+            "conflict_message":
+                "The rack" + deleted_relation_message + nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.related_decommissioned_asset:
+        conflicts.append({
+            "conflict_message":
+                "This asset has been decommissioned in a live change. " +
+                nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if len(conflicts) == 0:
+        if asset_cp.is_conflict:
+            conflicts.append({
+                "conflict_message":
+                    "Live changes have been made to this asset since your " +
+                    "latest change planner modification. This conflict " +
+                    "needs to be resolved. Please select which version " +
+                    "you would like to keep.",
+                "conflicting_asset": None,
+                "conflict_resolvable": True,
+            })
+    if len(conflicts) == 0:
+        return None
+    else:
+        return conflicts
+
+
+def get_modifications_in_cp(change_plan):
+    assets_cp = AssetCP.objects.filter(change_plan=change_plan)
+    modifications = []
+    for asset_cp in assets_cp:
+        assetCP_data = RecursiveAssetCPSerializer(asset_cp).data
+        related_asset = asset_cp.related_asset
+        if asset_cp.is_decommissioned:
+            title = "Decommission asset"
+            if asset_cp.asset_number:
+                title += " " + str(asset_cp.asset_number)
+            asset_data = None
+        elif related_asset:
+            title = "Modify asset " + str(asset_cp.related_asset.asset_number)
+            asset_data = RecursiveAssetSerializer(related_asset).data
+        else:
+            title = "Create asset"
+            if asset_cp.asset_number:
+                title += " " + str(asset_cp.asset_number)
+            asset_data = None
+        modifications.append({
+            "title": title,
+            "asset": asset_data,
+            "asset_cp": assetCP_data,
+            "conflicts": get_cp_modification_conflicts(asset_cp)
```

Comment:
can you also include `change typ` and `changes` field? as documented on the notion